cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -128,22 +130,31 @@ func newEtcdClient() (client *etcd.Client) {
 
 // TODO: evaluate using pkg/client/clientcmd
 func newKubeClient() (*kclient.Client, error) {
-	config := &kclient.Config{}
-
-	masterHost := os.Getenv("KUBERNETES_RO_SERVICE_HOST")
-	if masterHost == "" {
-		log.Fatalf("KUBERNETES_RO_SERVICE_HOST is not defined")
-	}
-	masterPort := os.Getenv("KUBERNETES_RO_SERVICE_PORT")
-	if masterPort == "" {
-		log.Fatalf("KUBERNETES_RO_SERVICE_PORT is not defined")
+	var config *kclient.Config
+	if *kubecfg_file == "" {
+		// No kubecfg file provided. Use kubernetes_ro service.
+		masterHost := os.Getenv("KUBERNETES_RO_SERVICE_HOST")
+		if masterHost == "" {
+			log.Fatalf("KUBERNETES_RO_SERVICE_HOST is not defined")
+		}
+		masterPort := os.Getenv("KUBERNETES_RO_SERVICE_PORT")
+		if masterPort == "" {
+			log.Fatalf("KUBERNETES_RO_SERVICE_PORT is not defined")
+		}
+		config = &kclient.Config{
+			Host:    fmt.Sprintf("http://%s:%s", masterHost, masterPort),
+			Version: "v1beta1",
+		}
+	} else {
+		var err error
+		if config, err = kclientcmd.NewNonInteractiveDeferredLoadingClientConfig(
```

Comment:
Changing to v1beta3 required other changes. I'd rather see that go in a different PR.